solidity-checks/src/main/java/org/sonarsource/solidity/checks/GuardCheckPatternCheck.java

```diff
@@ -0,0 +1,105 @@
+package org.sonarsource.solidity.checks;
+
+import java.util.List;
+import java.util.Optional;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.sonar.check.Rule;
+import org.sonarsource.solidity.frontend.SolidityParser.ExpressionContext;
+import org.sonarsource.solidity.frontend.SolidityParser.FunctionCallArgumentsContext;
+import org.sonarsource.solidity.frontend.SolidityParser.FunctionCallContext;
+import org.sonarsource.solidity.frontend.SolidityParser.FunctionDefinitionContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ModifierInvocationContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ModifierListContext;
+import org.sonarsource.solidity.frontend.SolidityParser.ParameterContext;
+import org.sonarsource.solidity.frontend.SolidityParser.SimpleStatementContext;
+import org.sonarsource.solidity.frontend.SolidityParser.StatementContext;
+
+@Rule(key = GuardCheckPatternCheck.RULE_KEY)
+public class GuardCheckPatternCheck extends IssuableVisitor {
+
+  public static final String RULE_KEY = "ExternalRule17";
+
+  @Override
+  public ParseTree visitFunctionDefinition(FunctionDefinitionContext ctx) {
+    ModifierListContext modifiersList = ctx.modifierList();
+    if (CheckUtils.isPublicOrExternalFunction(modifiersList)
+      && !CheckUtils.isViewOrPureFunction(modifiersList.stateMutability()) && ctx.block() != null) {
+      List<ParameterContext> parameters = ctx.parameterList().parameter();
+      parameters.stream()
+        .filter(parameter -> parameterIsNotGuardChecked(parameter.identifier().getText(), modifiersList.modifierInvocation(), ctx.block().statement()))
+        .forEach(parameter -> ruleContext().addIssue(parameter.getStart(), parameter.getStop(), parameter.getStop().getText().length(),
+          "You should check with require the validity of the parameter " + parameter.getChild(1).getText() + ".", RULE_KEY));
+    }
+    return super.visitFunctionDefinition(ctx);
+  }
+
+  private static boolean parameterIsNotGuardChecked(String parameter,
+    List<ModifierInvocationContext> modifierInvocationList, List<StatementContext> statementList) {
+    if (argumentCheckedInModifier(modifierInvocationList, parameter)) {
+      return false;
+    }
+    if (argumentCheckedInFunction(statementList, parameter)) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this if-then-else statement by a single return statement. ([squid:S1126](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1126))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=71&issues=AWTF-t3pxXt_iLiSnF6f&open=AWTF-t3pxXt_iLiSnF6f)

[ISSUE_KEY]: AWTF-t3pxXt_iLiSnF6f